utils/S3UtilsMongoClient.js

```diff
@@ -269,6 +374,138 @@ class S3UtilsMongoClient extends MongoClientInterface {
             return cb();
         });
     }
+
+    updateStorageConsumptionMetrics(countItems, dataMetrics, log, cb) {
+        const updatedStorageMetricsList = [{ _id: __COUNT_ITEMS, value: countItems }];
+        // iterate every resource through dataMetrics and add to updatedStorageMetricsList
+        Object.keys(dataMetrics).forEach(metricLevel => {
+            if (validStorageMetricLevels.has(metricLevel)) {
+                Object.keys(dataMetrics[metricLevel]).forEach(resource => {
+                    updatedStorageMetricsList.push({
+                        _id: `${metricLevel}:${resource}`,
+                        value: {
+                            measuredOn: new Date(),
+                            ...dataMetrics[metricLevel][resource],
+                        },
+                    });
+                });
+            }
+        });
+        let tempCollection;
+        async.series([
+            next => this.getCollection(INFOSTORE_TMP).drop(err => {
+                if (err && err.codeName !== 'NamespaceNotFound') {
+                    return next(err);
+                }
+                return next();
+            }),
+            next => this.db.createCollection(INFOSTORE_TMP, (err, collection) => {
+                if (err) {
+                    return next(err);
+                }
+                tempCollection = collection;
+                return next();
+            }),
+            next => tempCollection.insertMany(updatedStorageMetricsList, { ordered: false }, next),
+            next => async.retry(
+                3,
+                done => tempCollection.rename(INFOSTORE, { dropTarget: true }, done),
+                err => {
+                    if (err) {
+                        log.error('updateStorageConsumptionMetrics: error renaming temp collection, try again', {
+                            error: err.message,
+                        });
+                        return next(err);
+                    }
+                    return next();
+                },
+            ),
+        ], err => {
+            if (err) {
+                log.error('updateStorageConsumptionMetrics: error updating count items', {
+                    error: err.message,
+                });
+                return cb(errors.InternalError);
+            }
+            return cb();
+        });
+    }
```

Comment:
I tested the current count-items job in my cluster where I have thousands of buckets, but this logic:
- puts all the buckets name under `_id: countitems` (`bucketList`). I think this can be deleted, unless we need to expose the list of all buckets. Note that this may fail if the list is too large, as everything is in a single document.
- It only creates one document of each type. See below

Here is my (truncated) output:

```json
{ "_id" : "countitems", "value" : { "objects" : 1057057, "versions" : 0, "buckets" : 1911, "bucketList" : [ { "name" : "my-bucket-2019", "location" : "us-east-1", "isVersioned" : false, "ownerCanonicalId" : "7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd", "ingestion" : false }, { "name" : "test", "location" : "us-east-1", "isVersioned" : false, "ownerCanonicalId" : "7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd", "ingestion" : false }, { "name" : "my-bucket-1162", "location" : "us-east-1", "isVersioned" : false, "ownerCanonicalId" : "7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd", "ingestion" : false }, { "name" : "my-bucket-1246", "location" : "us-east-1", "isVersioned" : false, "ownerCanonicalId" : "7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd", "ingestion" : false }, { "name" : "my-bucket-1192", "loca.... [ all the buckets here) ],
{ "_id" : "location:fd2234c7-c258-11ed-aae3-ea8ec16f01ad", "value" : { "measuredOn" : ISODate("2023-03-31T06:55:20.786Z"), "usedCapacity" : { "current" : 0, "nonCurrent" : 0 }, "objectCount" : { "current" : 1057057, "nonCurrent" : 0, "deleteMarker" : 0 } } }
{ "_id" : "account:7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd", "value" : { "measuredOn" : ISODate("2023-03-31T06:55:20.786Z"), "usedCapacity" : { "current" : 0, "nonCurrent" : 0 }, "objectCount" : { "current" : 1057057, "nonCurrent" : 0, "deleteMarker" : 0 }, "locations" : { "fd2234c7-c258-11ed-aae3-ea8ec16f01ad" : { "usedCapacity" : { "current" : 0, "nonCurrent" : 0 }, "objectCount" : { "current" : 1057057, "nonCurrent" : 0, "deleteMarker" : 0 } } } } }
{ "_id" : "bucket:test_1680008330869", "value" : { "measuredOn" : ISODate("2023-03-31T06:55:20.786Z"), "usedCapacity" : { "current" : 0, "nonCurrent" : 0 }, "objectCount" : { "current" : 1057057, "nonCurrent" : 0, "deleteMarker" : 0 } } }
```

```bash
data-db-mongodb-sharded-shard-0:PRIMARY> db.getCollection('__infostore').find().count()
4
```

For some reason I see only one document of each type (like, only 1 `bucket:test_...`) while I have many others, I believe this is not on purpose.

Reply:
After adding some logs this logic seem fine but the function never receives all the resources for some reason

```
size of datametrics {
  bucket: {
    test_1680008330869: { usedCapacity: [Object], objectCount: [Object] }
  },
  location: {
    'fd2234c7-c258-11ed-aae3-ea8ec16f01ad': { usedCapacity: [Object], objectCount: [Object] }
  },
  account: {
    '7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd': {
      usedCapacity: [Object],
      objectCount: [Object],
      locations: [Object]
    }
  }
}
size of dataMetrics[metricLevel] {
  test_1680008330869: {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 }
  }
}
size of dataMetrics[metricLevel][resource] {
  test_1680008330869: {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 }
  }
}
size of dataMetrics[metricLevel] {
  'fd2234c7-c258-11ed-aae3-ea8ec16f01ad': {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 }
  }
}
size of dataMetrics[metricLevel][resource] {
  'fd2234c7-c258-11ed-aae3-ea8ec16f01ad': {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 }
  }
}
size of dataMetrics[metricLevel] {
  '7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd': {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 },
    locations: { 'fd2234c7-c258-11ed-aae3-ea8ec16f01ad': [Object] }
  }
}
size of dataMetrics[metricLevel][resource] {
  '7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd': {
    usedCapacity: { current: 0, nonCurrent: 0 },
    objectCount: { current: 1057057, nonCurrent: 0, deleteMarker: 0 },
    locations: { 'fd2234c7-c258-11ed-aae3-ea8ec16f01ad': [Object] }
  }
}
going to put 4 [
  {
    _id: 'countitems',
    value: {
      objects: 1057057,
      versions: 0,
      buckets: 1911,
      bucketList: [Array],
      dataManaged: [Object],
      stalled: 0
    }
  },
  {
    _id: 'bucket:test_1680008330869',
    value: {
      measuredOn: 2023-03-31T07:17:12.556Z,
      usedCapacity: [Object],
      objectCount: [Object]
    }
  },
  {
    _id: 'location:fd2234c7-c258-11ed-aae3-ea8ec16f01ad',
    value: {
      measuredOn: 2023-03-31T07:17:12.557Z,
      usedCapacity: [Object],
      objectCount: [Object]
    }
  },
  {
    _id: 'account:7a9747d06c80a1f046fdd134522699213df25632a7de36b55ee78d069e2425fd',
    value: {
      measuredOn: 2023-03-31T07:17:12.558Z,
      usedCapacity: [Object],
      objectCount: [Object],
      locations: [Object]
    }
  }
]
```

![image](https://user-images.githubusercontent.com/91462779/229050726-9b42aefa-9944-4d7b-8476-e5b591953ded.png)

Reply:
Closing this comment